### PR TITLE
Add PeerChainState

### PR DESCRIPTION
**Pull request by earlbread**
This adds `PeerChainStatus` struct and `Swarm<T>.GetPeerChainStatus()` method to check the chain status of the connected peers.

### CODE REVIEW

**Comment by earlbread** [comment]
I changed an `Array` to a `HashSet` to fix the flaky test. 945d6d3

File: Libplanet/Net/Swarm.cs [hunk 1]
```diff
@@ -404,6 +404,26 @@ public string TraceTable()
...
+        /// <summary>
+        /// Gets the <see cref="PeerChainStatus"/> of the connected <see cref="Peers"/>.
+        /// </summary>
+        /// <param name="dialTimeout">A timeout value for dialing.</param>
+        /// <param name="cancellationToken">
+        /// A cancellation token used to propagate notification that this
+        /// operation should be canceled.
+        /// </param>
+        /// <returns><see cref="PeerChainStatus"/> of the connected <see cref="Peers"/>.</returns>
+        public async Task<IEnumerable<PeerChainStatus>> GetPeerChainStatus(
```
Comment (dahlia) [inline Libplanet/Net/Swarm.cs:416]: The name should end with `Async`.

Comment (earlbread) [reply Libplanet/Net/Swarm.cs]: Renamed `GetPeerChainStatus` to `GetPeerChainStatusAsync`. ba45953

**Comment by earlbread** [comment]
> How about renaming `PeerChainStatus` to `PeerChainState`? Cf. https://english.stackexchange.com/questions/12958/status-vs-state. /cc @planetarium/libplanet

I think that's good. I'll rename it.

**Comment by limebell** [comment]
Currently, `Swarm<T>.DialToExistingPeers` returns `Task<(BoundPeer, ChainStatus)>` as its return type. How about merge `PeerChainState` struct and `ChainStatus` (and make it public)? Then we can just use `Task<PeerChainState>` as the method's return type.

File: Libplanet.Tests/Net/SwarmTest.cs [hunk 2]
```diff
@@ -2016,6 +2016,58 @@ public async Task DoNotFillMultipleTimes()
             }
         }
 
+        [Fact(Timeout = Timeout)]
+        public async Task GetPeerChainStatusAsync()
```
Comment (dahlia) [inline Libplanet.Tests/Net/SwarmTest.cs:2020]:
```suggestion
        public async Task GetPeerChainStateAsync()
```

**Comment by earlbread** [comment]
> Currently, `Swarm<T>.DialToExistingPeers` returns `Task<(BoundPeer, ChainStatus)>` as its return type. How about merge `PeerChainState` struct and `ChainStatus` (and make it public)? Then we can just use `Task<PeerChainState>` as the method's return type.

I thought about it at first, but to do that, the current internal `Message` class should be public. Is it okay?

**Comment by limebell** [comment]
> > Currently, `Swarm<T>.DialToExistingPeers` returns `Task<(BoundPeer, ChainStatus)>` as its return type. How about merge `PeerChainState` struct and `ChainStatus` (and make it public)? Then we can just use `Task<PeerChainState>` as the method's return type.
> 
> I thought about it at first, but to do that, the current internal `Message` class should be public. Is it okay?

OH I made mistake, you're right.

**Pull request merged**